src/farm/library.cairo

```diff
@@ -547,4 +607,134 @@ namespace CarbonableFarmer {
+                token_total_deposited=token_total_deposited,
+                total_amount=total_amount,
+                address_token_amount=address_token_amount,
+            );
+
+            // [Interaction] Starkvest - create vesting for address
+            // Vesting, with no cliff period, no duration and no delay to start
+            let beneficiary = address;
+            let cliff_delta = 0;
+            let start = 1;
+            let duration = 1;
+            let slice_period_seconds = 1;
+            let revocable = TRUE;
+
+            // [Interaction] Starkvest - create vesting for address
+            let (vesting_id) = IStarkVest.create_vesting(
+                starkvest_address,
+                beneficiary,
+                cliff_delta,
+                start,
+                duration,
+                slice_period_seconds,
+                revocable,
+                address_amount,
+            );
+
+            // TODO: verify vesting_id is not 0
+
+            // [Event] Emit addr vesting are created
+            let (current_time) = get_block_timestamp();
+            VestingOfAddrCreated.emit(address=address, vesting_id=vesting_id, time=current_time);
+
+            tempvar _syscall_ptr = syscall_ptr;
+            tempvar _pedersen_ptr = pedersen_ptr;
+            tempvar _range_check_ptr = range_check_ptr;
+        } else {
+            tempvar _syscall_ptr = syscall_ptr;
+            tempvar _pedersen_ptr = pedersen_ptr;
+            tempvar _range_check_ptr = range_check_ptr;
+        }
+        let syscall_ptr = _syscall_ptr;
+        let pedersen_ptr = _pedersen_ptr;
+        let range_check_ptr = _range_check_ptr;
+
+        let zero = Uint256(low=0, high=0);
+        let (is_zero) = uint256_eq(index, zero);
+        // Stop recursion if index is 0
+        if (is_zero == TRUE) {
+            return ();
+        }
+        // Else move on to the next index
+        let (next) = SafeUint256.sub_le(index, one);
+        return _create_vestings_iter(
+            contract_address=contract_address,
+            starkvest_address=starkvest_address,
+            total_amount=total_amount,
+            token_total_deposited=token_total_deposited,
+            index=next,
+            total_supply=total_supply,
+        );
+    }
+
+    // We have to make a cross product, between address_token_amount, total_amount and token_total_deposited
+    // to find the amount to distribut
+    func _amount_to_vesting{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        token_total_deposited: Uint256, total_amount: felt, address_token_amount: felt
+    ) -> (amount: Uint256) {
+        let dividend = total_amount * address_token_amount;
+        let dividend_uint256 = Uint256(low=dividend, high=0);
+
+        // [Check] Uint256 compliance
+        with_attr error_message("CarbonableFarmer: dividend_uint256 is not a valid Uint256") {
+            uint256_check(dividend_uint256);
```

Comment:
How large can `dividend` be?
This will result in a range_check on dividend, so we need to make sure the product is not too large (an doesn't overflow)

Reply:
I will create an Issue dedicated on this point, with label security

Reply:
Issue created #107 : Implement control to NOT overflow the variable dividend and dividend_uint256